### PR TITLE
[WIP] Use out-of-tree k8s go client.

### LINKED ISSUE #38: Consider using client-go instead of the unversioned client.
Right now NFD uses the `unversioned` client to talk to the K8s cluster. We should consider using the `client-go` (https://github.com/kubernetes/client-go) instead.

### PR DESCRIPTION
**Pull request by ConnorDoyle**
Fixes #38.

Builds and tests pass, but still needs manual testing (since we don't have e2e set up yet.)

This also speeds up the build quite a bit.

### CODE REVIEW

**Comment by ConnorDoyle** [comment]
Manual testing in our bare metal dev cluster worked. Think this is OK to merge after your LGTM @balajismaniam

File: main.go [hunk 1]
```diff
@@ -8,8 +8,9 @@ import (
 	"strings"
 
 	"github.com/docopt/docopt-go"
-	"k8s.io/kubernetes/pkg/api"
-	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/client-go/kubernetes"
```
Comment (balajismaniam) [inline main.go]: Can we alias this as `client` or `k8sclient`?

File: main.go [hunk 2]
```diff
@@ -8,8 +8,9 @@ import (
...
+	api "k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/rest"
```
Comment (balajismaniam) [inline main.go]: Alias this as `restclient`?

File: main.go (hunk 1 above)
Comment (ConnorDoyle) [inline main.go]: Will update.

File: main.go (hunk 2 above)
Comment (ConnorDoyle) [inline main.go]: Will update.

Comment (ConnorDoyle) [inline main.go]: updated

**Pull request merged**